### ISSUE TITLE
Inconsistent TOTP code generation

### ISSUE THREAD

**Issue by blackheaven** [haskell-cryptography/one-time-password#5]
After a talk with @Kleidukos I have started to figure-out why TOTP implementation was not working with some TOTP Apps.

I have tested it with other TOTP libraries and I have found-out:

* HOTP SHA1 implementation differs on some other implementations (maybe reference table should be in tests)
* All TOTP are differents

Testing code

```python
>>> import pyotp
>>> bkey = '5EGLVYWX2GD7MFEAMND47V2QAK6Q3OCHIUIQSWM5UUD6RWUIX5BQ====' # base32 of e90cbae2d7d187f614806347cfd75002bd0db847451109599da507e8da88bf43
>>> pyotp.totp.TOTP(bkey, digest=pyotp.hashlib.sha1).at(1286057400)
'067007'
>>> pyotp.totp.TOTP(bkey, digest=pyotp.hashlib.sha256).at(1286057400)
'408397'
>>> pyotp.totp.TOTP(bkey, digest=pyotp.hashlib.sha512).at(1286057400)
'594741'
```

**Comment by Kleidukos** [comment]
Taking the reference table of the HOTP RFC is not possible because we enforce a 32 bytes length for the key size and the key shown in the test vectors is 20-bytes long.

However we can build a reference table from the python library, that's entirely reasonable.

**Comment by Kleidukos** [comment]
Fixed by 2ca23134bab22fc8a1bea775eebfcc6cde11ce14
